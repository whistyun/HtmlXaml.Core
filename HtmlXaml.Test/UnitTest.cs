
using ApprovalTests;
using ApprovalTests.Reporters;
using HtmlXaml.Core;
using NUnit.Framework;
using System.Linq;
using System.Threading;
using System.Windows.Documents;

namespace HtmlXaml.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest
    {
        [Test]
        [Apartment(ApartmentState.STA)]
        public void Button()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void CodeBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void InlineCode()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Input()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void List()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Progres()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalInline()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
