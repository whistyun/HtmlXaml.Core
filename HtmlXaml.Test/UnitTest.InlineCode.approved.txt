<?xml version="1.0" encoding="utf-16"?>
<FlowDocument xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <Paragraph>An <Run Tag="Code">apple</Run> is an editable fruit produced by an <Run Tag="Code">apple tree</Run>.</Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <Run Tag="Code">Apple trees</Run> are caltivated wordwide.</Paragraph>
</FlowDocument>
